runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -750,6 +750,8 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, job
 
 				/* Check for resolution exception */
 				if (VM_VMHelpers::exceptionPending(currentThread)) {
+					j9mem_free_memory(signature);
+					j9mem_free_memory(name);
```

Comment:
There are now two places that free these variables - here and line ~245.  If you move the code to free them to after the `done:` label, there will be 1 place making it easier to reason about the allocation lifetimes.  This works becuase it's safe to free `null`